src/main/java/ch/njol/skript/lang/SyntaxElementInfo.java

```diff
@@ -30,8 +31,11 @@ public class SyntaxElementInfo<E extends SyntaxElement> {
 	public final Class<E> elementClass;
 	public final String[] patterns;
 	public final String originClassPath;
-
+  
 	public SyntaxElementInfo(String[] patterns, Class<E> elementClass, String originClassPath) throws IllegalArgumentException {
+		if (Modifier.isAbstract(elementClass.getModifiers()))
+			throw new IllegalArgumentException("Class " + elementClass.getName() + " is abstract");
```

Comment:
i think this would be a skriptapiexception since it is the api being used incorrectly